src/lib/database.js

```diff
@@ -61,11 +55,19 @@ class FileConnection extends Connection {
     const filePath = path.resolve(config.data.directory, name + '.json');
     super(Klass.deserialize(readJSONFile(filePath)), gcInterval);
     this.filePath = filePath;
+    this.saveInterval = setInterval(() => {
+      this.save().catch(console.error);
+    }, config.data.saveInterval);
+  }
+
+  save() {
+    const data = this.db.serialize();
+    return fs.writeJson(this.filePath, data);
   }
 
   close() {
-    super.close();
-    writeJSONFile(this.filePath, this.db.serialize());
+    clearInterval(this.saveInterval);
+    return Promise.all([super.close(), this.save()]);
```

Comment:
Why forcing `super.close` to be a promise and returning `Promise.all` instead of first `super.close()` then `return this.save()`?
Future-proofing if close becomes async?

Reply:
It's because if we're Connection/InMemoryConnection, we want also to get a Promise from close().

Reply:
Ok, sounds good then! 👍